internal/translation/datafederation/conversion_test.go

```diff
@@ -0,0 +1,46 @@
+package datafederation
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	fuzz "github.com/google/gofuzz"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+)
+
+func TestRoundtrip_DataFederation(t *testing.T) {
+	f := fuzz.New()
+
+	for i := 0; i < 100; i++ {
+		var atlas admin.DataLakeTenant
+		f.Fuzz(&atlas)
```

Comment:
[side comment]
It is a pity that current stdlib fuzz does not do structs. This is way simpler to write than having to spell out all the fields.

Still it is surprising this does not hit some ilegal values. Sometimes you might want fuzz to find those, other the ilegal values are just problematic or already render explicit errors in code.

Reply:
well, the fuzzer did find some invalid values. It's just that I fixed them to get it green 😆